### ISSUE TITLE
Weird error ggml_metal_graph_compute: command buffer 0 failed with status 0 on Apple M2 Pro

### ISSUE THREAD

**Issue by jooray** [ggerganov/whisper.cpp#1516]
I just built current version of whisper.cpp after git pull and tried transcription. Got:

```
ggml_metal_init: maxTransferRate               = built-in GPU
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =  3117.88 MB, ( 3118.53 / 77309.41)
whisper_model_load:    Metal buffer size =  3117.87 MB
whisper_model_load: model size    = 3117.39 MB
whisper_backend_init: using Metal backend
ggml_metal_init: allocating
ggml_metal_init: found device: Apple M2 Max
ggml_metal_init: picking default device: Apple M2 Max
ggml_metal_init: default.metallib not found, loading from source
ggml_metal_init: loading '/Users/user/whisper.cpp/ggml-metal.metal'
ggml_metal_init: GPU name:   Apple M2 Max
ggml_metal_init: GPU family: MTLGPUFamilyApple8 (1008)
ggml_metal_init: hasUnifiedMemory              = true
ggml_metal_init: recommendedMaxWorkingSetSize  = 77309.41 MB
ggml_metal_init: maxTransferRate               = built-in GPU
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =   220.20 MB, ( 3338.73 / 77309.41)
whisper_init_state: kv self size  =  220.20 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =   245.76 MB, ( 3584.49 / 77309.41)
whisper_init_state: kv cross size =  245.76 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =     0.02 MB, ( 3584.51 / 77309.41)
whisper_init_state: compute buffer (conv)   =   32.36 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =     0.02 MB, ( 3584.52 / 77309.41)
whisper_init_state: compute buffer (encode) =  212.36 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =     0.02 MB, ( 3584.54 / 77309.41)
whisper_init_state: compute buffer (cross)  =    9.32 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =     0.02 MB, ( 3584.56 / 77309.41)
whisper_init_state: compute buffer (decode) =   99.17 MB
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =    30.72 MB, ( 3615.28 / 77309.41)
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =   210.73 MB, ( 3826.01 / 77309.41)
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =     7.68 MB, ( 3833.69 / 77309.41)
ggml_metal_add_buffer: allocated 'backend         ' buffer, size =    97.53 MB, ( 3931.23 / 77309.41)

system_info: n_threads = 4 / 12 | AVX = 0 | AVX2 = 0 | AVX512 = 0 | FMA = 0 | NEON = 1 | ARM_FMA = 1 | METAL = 1 | F16C = 0 | FP16_VA = 1 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 0 | SSSE3 = 0 | VSX = 0 | CUDA = 0 | COREML = 0 | OPENVINO = 0 |

main: processing 'source.wav' (4739448 samples, 296.2 sec), 4 threads, 1 processors, 5 beams + best of 5, lang = auto, task = transcribe, timestamps = 1 ...

ggml_metal_graph_compute: command buffer 0 failed with status 0
GGML_ASSERT: ggml-metal.m:1611: false
```

**Comment by marekhac** [comment]
Got the same issue on MacBookPro (2019). Is there any solution for that ?

**Comment by jooray** [comment]
git pull and recompile seems to have solved that.

**Comment by marekhac** [comment]
> git pull and recompile seems to have solved that.

Still doesn't work on my Intel based MacBookPro. But today I've tested it on a new MacBookPro with M2 Pro processor, and it works fine.

**Comment by ggerganov** [comment]
On Intel macs you have to disable the GPU - add `--no-gpu`

**Comment by marekhac** [comment]
> On Intel macs you have to disable the GPU - add `--no-gpu`

Yes, it solved the problem. Thanks!

**Comment by war5** [comment]
I had this same issue on an M1 macbook air when using the `samples/jfk.wav`. 

I solved this by updating OS X to 14.2 from 13.X. When I rebuilt the sample worked correctly.
